sourcegen-bytecode-writer/src/main/java/io/micronaut/sourcegen/bytecode/expression/InvokeInstanceMethodExpressionWriter.java

```diff
@@ -53,8 +54,14 @@ public void write(GeneratorAdapter generatorAdapter, MethodContext context) {
         Type methodOwnerType = TypeUtils.getType(instanceType, context.objectDef());
         MethodDef methodDef = invokeInstanceMethod.method();
         Method method = new Method(methodDef.getName(), TypeUtils.getMethodDescriptor(context.objectDef(), methodDef));
-        if (instanceType instanceof TypeDef.TypeVariable typeVariable) {
-            instanceType = typeVariable.bounds().get(0);
+        while (instanceType instanceof TypeDef.TypeVariable typeVariable) {
```

Comment:
It fixes https://github.com/micronaut-projects/micronaut-security/issues/1981.

What is strange is that the type being called is actually
```java
TypeVariable[
  name=V, 
  bounds=[
     TypeVariable[
       name=V,
       bounds=[
         ClassElementType[
           classElement=java.lang.Object, 
           nullable=false
         ]
       ], 
       nullable=false
     ], 
     ClassElementType[
       classElement=java.lang.Object, 
       nullable=false
     ]
  ], 
  nullable=false
]
```

So it is like saying `V extends (V extends Object)`